packages/web/src/utils/menus/utils.ts

```diff
@@ -0,0 +1,198 @@
+/**
+* Handle event for keyboard control of menu
+* @param {Event} event
+* @param {Element} menu
+* @param {any[]} queue
+*/
+export async function handleKeyDownMenu(event, menu, queue) {
+  let key = event.keyCode || event.which;
+  let currentIndex = queue.indexOf(document.activeElement);
+  let activeElement = queue[currentIndex];
+
+  switch(key) {
+    // Down arrow
+    case 40:
+      event.preventDefault();
+      // If on last item, jump to first item
+      if (currentIndex + 1 > queue.length - 1) {
+        await focusElement(0, queue);
+      // Jump to next item
+      } else {
+        await focusElement(currentIndex + 1, queue);
+      }
+      break;
+
+    // Up arrow
+    case 38:
+      event.preventDefault();
+      // If on first item, jump to last item
+      if (currentIndex - 1 < 0) {
+        await focusElement(queue.length - 1, queue);
+      // Jump to previous item
+      } else {
+        await focusElement(currentIndex - 1, queue);
+      }
+      break;
+
+    // Right arrow
+    case 39:
+      event.preventDefault();
+      // If current item is group-menu trigger
+      if (activeElement.nodeName == "GCDS-MENU-GROUP" && !activeElement.hasAttribute("open")) {
+        await openMenuGroup(activeElement as HTMLGcdsMenuGroupElement, menu);
```

Comment:
I think you should be able to just use `activeElement ` instead of `activeElement as HTMLGcdsMenuGroupElement` because you are already checking the node name before using the element.